content/articles/2019-05-vroom-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,450 @@
+```
+
+- Use numeric indices
+```{r}
+data <- vroom("flights.tsv", col_select = c(1, 3, 11))
+```
+- Use the selection helpers
+```{r}
+data <- vroom("flights.tsv", col_select = ends_with("time"))
+```
+- Or rename columns
+```{r}
+data <- vroom("flights.tsv", col_select = list(plane = tailnum, everything()))
+data
+```
+
+## Name repair
+
+Often the names of columns in the original dataset are not ideal to work with.
+`vroom()` uses the same [.name_repair](https://www.tidyverse.org/articles/2019/01/tibble-2.0.1/#name-repair)
+argument as tibble, so you can use one of the default name repair strategies or
+provide a custom function. A great approach is to use the
+[janitor](http://sfirke.github.io/janitor/) `make_clean_names()` function as the input.
+
+```{r}
+vroom("flights.tsv", .name_repair = janitor::make_clean_names)
+
+vroom("flights.tsv", .name_repair = ~ janitor::make_clean_names(., case = "lower_camel"))
+```
+
+
+## Column types
+
+Like readr, vroom guesses the data types of columns as they are read, however sometimes it
+is necessary to change the type of one or more columns.
+
+The available specifications are: (with single letter abbreviations in quotes)
+
+* `col_logical()` 'l', containing only `T`, `F`, `TRUE`, `FALSE`, `1` or `0`.
+* `col_integer()` 'i', integer values.
+* `col_double()` 'd', floating point values.
+* `col_number()` [n], numbers containing the `grouping_mark`
+* `col_date(format = "")` [D]: with the locale's `date_format`.
+* `col_time(format = "")` [t]: with the locale's `time_format`.
+* `col_datetime(format = "")` [T]: ISO8601 date times.
+* `col_factor(levels, ordered)` 'f', a fixed set of values.
+* `col_character()` 'c', everything else.
+* `col_skip()` '_, -', don't import this column.
+* `col_guess()` '?', parse using the "best" type based on the input.
+
+You can tell vroom what columns to use with the `col_types()` argument in a number of ways.
+
+If you only need to override a single column the most concise way is to use a named vector.
+
+```{r}
+# read the 'year' column as an integer
+data <- vroom("flights.tsv", col_types = c(year = "i"))
+
+# also skip reading the 'time_hour' column
+data <- vroom("flights.tsv", col_types = c(year = "i", time_hour = "_"))
+
+# also read the carrier as a factor
+data <- vroom("flights.tsv", col_types = c(year = "i", time_hour = "_", carrier = "f"))
+```
+
+However you can also use the `col_*()` functions in a list.
+
+```{r}
+data <- vroom("flights.tsv",
+  col_types = list(year = col_integer(), time_hour = col_skip(), carrier = col_factor())
+)
+```
+
+This is most useful when a column type needs additional information, such as
+for categorical data when you know all of the levels of a factor.
+
+```{r}
+data <- vroom("flights.tsv",
+  col_types = list(dest = col_factor(levels = c("EWR", "JFK", "LGA")))
+)
+```
+
+## Speed
+
+vroom is fast, but how fast?
+We benchmarked vroom using a real world dataset of taxi trip data, with
```

Comment:
real-world gets hyphenated here, since it's modifying dataset (technically taxi-trip, too, but dealer's choice there)